packages/react-components/react-button/src/components/Button/useButtonStyles.styles.ts

```diff
@@ -332,6 +332,9 @@ const useRootDisabledStyles = makeStyles({
       [`& .${iconRegularClassName}`]: {
         display: 'inline',
       },
+      [`&:disabled .${buttonClassNames.icon}`]: {
+        color: tokens.colorNeutralForegroundDisabled,
+      },
```

Comment:
For completeness you should probably also add this same style in the base (around line 321 below the `cursor: 'not-allowed'` line) and `:hover:active` states (around line 352 similar to where you're putting this one but for `:hover:active`).

Another thing is that you don't really need the `disabled` style here (I doubled checked locally).

```suggestion
      [`& .${buttonClassNames.icon}`]: {
        color: tokens.colorNeutralForegroundDisabled,
      },
```